app/PaymentDrivers/RotessaPaymentDriver.php

```diff
@@ -0,0 +1,274 @@
+<?php
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2023. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+namespace App\PaymentDrivers;
+
+use Omnipay\Omnipay;
+use App\Models\Client;
+use App\Models\Payment;
+use App\Models\SystemLog;
+use App\Models\PaymentHash;
+use Illuminate\Support\Arr;
+use App\Models\GatewayType;
+use Omnipay\Rotessa\Gateway;
+use App\Models\ClientContact;
+use App\Utils\Traits\MakesHash;
+use App\Jobs\Util\SystemLogger;
+use App\PaymentDrivers\BaseDriver;
+use App\Models\ClientGatewayToken;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Database\Eloquent\Builder;
+use App\PaymentDrivers\Rotessa\Resources\Customer;
+use App\PaymentDrivers\Rotessa\PaymentMethod as Acss;
+use App\PaymentDrivers\Rotessa\PaymentMethod as BankTransfer;
+
+class RotessaPaymentDriver extends BaseDriver
+{
+    use MakesHash;
+
+    public $refundable = false;
+
+    public $token_billing = true;
+
+    public $can_authorise_credit_card = true;
+
+    public Gateway $gateway;
+
+    public $payment_method;
+
+    public static $methods = [
+        GatewayType::BANK_TRANSFER => BankTransfer::class,
+        //GatewayType::BACS => Bacs::class,
+        GatewayType::ACSS => Acss::class,
+        // GatewayType::DIRECT_DEBIT => DirectDebit::class
+    ];
+
+    public function init(): self
+    {
+       
+        $this->gateway = Omnipay::create(
+            $this->company_gateway->gateway->provider
+        );
+        $this->gateway->initialize((array) $this->company_gateway->getConfig());
+        return $this;
+    }
+
+    public function gatewayTypes(): array
+    {
+        $types = [];
+
+        if ($this->client
+        && $this->client->currency()
+        && in_array($this->client->currency()->code, ['USD'])
+        && isset($this->client->country)
+        && in_array($this->client->country->iso_3166_2, ['US'])) {
+            $types[] = GatewayType::BANK_TRANSFER;
+        }
+
+        if ($this->client
+            && $this->client->currency()
+            && in_array($this->client->currency()->code, ['CAD'])
+            && isset($this->client->country)
+            && in_array($this->client->country->iso_3166_2, ['CA'])) {
+            $types[] = GatewayType::ACSS;
+        }
+        
+        return $types;
+    }
```

Comment:
Please ensure that the supported payment methods applicable for this implementation are curated here.

ie. CAD + CAN only payments.

Reply:
Updated based on recommendations